src/Raven.Server/Rachis/Leader.cs

```
@@ -748,14 +749,22 @@ public void Dispose()
                     {
                         if (await waitAsync == false)
                         {
-                            if (rachisMergedCommand.Consumed.Raise())
+                            commandAborted = rachisMergedCommand.Consumed.Raise();
+                            if (commandAborted)
                                 throw new TimeoutException($"Waited for {timeout} but the command was not applied in this time.");
 
                             // if the command is already dequeued we must let it continue to keep its context valid.
                             await rachisMergedCommand.Tcs.Task;
                         }
                     }
                 }
+                catch (Exception) when (rachisMergedCommand.Consumed.Raise() == false && commandAborted == false && rachisMergedCommand.Tcs.Task.IsCompleted == false)
+                {
+                    // If the command has already been dequeued, we must allow it to continue to maintain its context validity.  
+                    // To avoid excessive iterations, if the lock is not acquired, we can simply wait for the task to complete.
+                    if (lockTaken == false)
+                        await rachisMergedCommand.Tcs.Task;
```

You are swallowing the exception here, should you?
This is when we dispose, so we need to clear the entire queue, sure, but if there is an error in here, what happens?
Note that if you have an exception, you'll have this raised _twice_, and then exit without waiting for the _next_ task.

---

> You are swallowing the exception here, should you?

As long as it is not an exception from a command task, yes, I want to swallow it and keep waiting.
An exception that comes from the command task, it will not be caught and will be thrown.

> This is when we dispose, so we need to clear the entire queue, sure, but if there is an error in here, what happens?

I didn't think of clearing the queue.
I want to keep waiting.
It can be a transient error (like out of memory).
Anyway, it comes to handle exceptions not coming from the command execution but from anything except that.

> Note that if you have an exception, you'll have this raised twice, and then exit without waiting for the next task.

Are you talking about rachisMergedCommand.Consumed.Raise()?
I try to abort the command while checking if it was consumed already.
I am not sure if that is what you are referring to.

---

The issue is if you have _two_ commands that you are running, and then you may still leave this early